src/System.Security.Cryptography.Xml/tests/EncryptionPropertyCollectionTest.cs

```diff
@@ -0,0 +1,192 @@
+            EncryptionProperty encProperty = new EncryptionProperty();
+            Assert.Equal(-1, encPropertyCollection.IndexOf(encProperty));
+        }
+
+        [Fact]
+        public void Insert_EmptyCollection()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty = new EncryptionProperty();
+            encPropertyCollection.Insert(0, encProperty);
+            Assert.Equal(0, encPropertyCollection.IndexOf(encProperty));
+        }
+
+        [Fact]
+        public void Insert_Beginning()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty1 = new EncryptionProperty();
+            EncryptionProperty encProperty2 = new EncryptionProperty();
+            EncryptionProperty encProperty3 = new EncryptionProperty();
+            encPropertyCollection.Add(encProperty1);
+            encPropertyCollection.Add(encProperty2);
+            encPropertyCollection.Insert(0, encProperty3);
+            Assert.Equal(0, encPropertyCollection.IndexOf(encProperty3));
+        }
+
+        [Fact]
+        public void Insert_Middle()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty1 = new EncryptionProperty();
+            EncryptionProperty encProperty2 = new EncryptionProperty();
+            EncryptionProperty encProperty3 = new EncryptionProperty();
+            encPropertyCollection.Add(encProperty1);
+            encPropertyCollection.Add(encProperty2);
+            encPropertyCollection.Insert(1, encProperty3);
+            Assert.Equal(1, encPropertyCollection.IndexOf(encProperty3));
+        }
+
+        [Fact]
+        public void Insert_End()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty1 = new EncryptionProperty();
+            EncryptionProperty encProperty2 = new EncryptionProperty();
+            EncryptionProperty encProperty3 = new EncryptionProperty();
+            encPropertyCollection.Add(encProperty1);
+            encPropertyCollection.Add(encProperty2);
+            encPropertyCollection.Insert(2, encProperty3);
+            Assert.Equal(2, encPropertyCollection.IndexOf(encProperty3));
+        }
+
+        [Fact]
+        public void Remove()
+        {
+            EncryptionPropertyCollection encPropertyCollection = new EncryptionPropertyCollection();
+            EncryptionProperty encProperty = new EncryptionProperty();
+            encPropertyCollection.Add(encProperty);
+            encPropertyCollection.Remove(encProperty);
+            Assert.Equal(-1, encPropertyCollection.IndexOf(encProperty));
+        }
```

Comment:
What about remove when not present?
What about remove null? (Or add null, or insert null, or removeat null)

For remove when present, consider enhancing it to have multiple items in the collection and verify that remove only changes the count by 1.  (Interesting thought, what if the same item is added multiple times and removed... is the count still only affected by 1?)

```C#
[Theory]
[InlineData(1, 0)]
[InlineData(5, 2)]
public void Remove(int collectionSize, int removeIdx)
{
    EncryptionPropertyCollection coll = new EncryptionPropertyCollection();
    EncryptionProperty toRemove = null;

    for (int i = 0; i < collectionSize; i++)
    {
        EncryptionProperty property = new EncryptionProperty();
        coll.Add(property);

        if (i == removeIdx)
        {
            toRemove = property;
        }
    }

    coll.Remove(toRemove);
    Assert.Equal(-1, coll.IndexOf(toRemove));
    Assert.Equal(collectionSize - 1, coll.Count);
}
```